### PR TITLE
GAWB-1236: Prevent single-owner acl from removing only owner permissions

### PR DESCRIPTION
**Pull request by rushtong**
PR for https://broadinstitute.atlassian.net/browse/GAWB-1236

This addresses, but does not complete GAWB-1236
Please see https://github.com/broadinstitute/agora/pull/171 for the Agora side of this PR. These can be tested separately.

Add a check for deletion of permissions to prevent removing all managers and therefore leaving the entity in an un-manageable state.

Looking for feedback on the UI handling. The one thing I don't like is the following condition: 
* if there is a single owner ACL, then that ACL is appropriately disabled
* upon adding new ACLs, or changing ACLs, that disabled state should toggle off if another owner is added.

---
- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [x] Tell the tech lead (TL) that the PR exists if they wants to look at it
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [x] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/cljs/org/broadinstitute/firecloud_ui/page/method_repo/methods_configs_acl.cljs [hunk 1]
```diff
@@ -25,6 +25,8 @@
 
 (def ^:private column-width "calc(50% - 4px)")
 
+(defn count-owners [acl-vec]
+  (count (filter #(= (:role %) owner-level) acl-vec)))
```
Comment (MattPutnam) [inline src/cljs/org/broadinstitute/firecloud_ui/page/method_repo/methods_configs_acl.cljs]: Another option:

```
(defn count-owners [acl-vec]
  (count (filter (comp #{owner-level} :role) acl-vec)))
```

**Review by MattPutnam — COMMENTED** [review]
Ugh... there are a bunch of things that are annoying me in this file.  Let me make a cleanup branch off this.

**Comment by MattPutnam** [comment]
👍 
Up to you to decide if the thing I suggested above is too arcane or not.

**Comment by rushtong** [comment]
It makes sense - I'll push an update out.

**Comment by davidangb** [comment]
There's a clickpath where the UI protection doesn't work:

1. Start editing an ACL
2. Add a new row. Leave the new row's user empty, but set the new row to OWNER.
3. Remove all previous owners.
4. Save.

This clickpath should still be protected on the API side, so the data is safe; but if we're going to add UI protection it shouldn't be this easy to defeat. I also don't know how to get around this ... it would be just as easy to add a new row and fill in a dummy email address, and without sending to the back end we can't know if that email address is valid or not.

**Comment by rushtong** [comment]
With the new requirements, there is no need to update the UI. Current errors from the API are correctly percolated up to the editing modal.

**Pull request closed without merging**